libshaderc_spvc/src/spvc.cc

```diff
@@ -515,7 +625,240 @@ shaderc_spvc_status shaderc_spvc_needs_buffer_size_buffer(
 void shaderc_spvc_build_combined_image_samplers(
     const shaderc_spvc_context_t context) {
   context->cross_compiler->build_combined_image_samplers();
-  return;
+}
+
+shaderc_spvc_status shaderc_spvc_get_execution_model(
+    const shaderc_spvc_context_t context,
+    shaderc_spvc_execution_model* execution_model) {
+  if (!context) return shaderc_spvc_status_missing_context_error;
+
+  if (!context->cross_compiler) {
+    context->messages.append(
+        "Invoked get_execution_model without an initialized compiler\n");
+    return shaderc_spvc_status_uninitialized_compiler_error;
+  }
+
+  if (!execution_model) {
+    context->messages.append(
+        "Invoked get_execution_model without a valid out param\n");
+    return shaderc_spvc_status_invalid_out_param;
+  }
+
+  auto spirv_model = context->cross_compiler->get_execution_model();
+  *execution_model = spv_model_to_spvc_model(spirv_model);
+  if (*execution_model == shaderc_spvc_execution_model_invalid) {
+    context->messages.append(
+        "Shader execution model appears to be of an unsupported type\n");
+    return shaderc_spvc_status_internal_error;
+  }
+
+  return shaderc_spvc_status_success;
+}
+
+shaderc_spvc_status shaderc_spvc_get_push_constant_buffer_count(
+    const shaderc_spvc_context_t context, size_t* count) {
+  if (!context) return shaderc_spvc_status_missing_context_error;
+
+  if (!context->cross_compiler) {
+    context->messages.append(
+        "Invoked push_constant_buffer_count without an initialized compiler\n");
+    return shaderc_spvc_status_uninitialized_compiler_error;
+  }
+
+  if (!count) {
+    context->messages.append(
+        "Invoked push_constant_buffer_count without a valid out param\n");
+    return shaderc_spvc_status_invalid_out_param;
+  }
+
+  *count = context->cross_compiler->get_shader_resources()
+               .push_constant_buffers.size();
+  return shaderc_spvc_status_success;
+}
+
+shaderc_spvc_status shaderc_spvc_get_binding_info(
+    const shaderc_spvc_context_t context, shaderc_spvc_shader_resource resource,
+    shaderc_spvc_binding_type binding_type, shaderc_spvc_binding_info* bindings,
+    size_t* binding_count) {
+  if (!context) return shaderc_spvc_status_missing_context_error;
+
+  if (!context->cross_compiler) {
+    context->messages.append(
+        "Invoked get_binding_info without an initialized compiler\n");
+    return shaderc_spvc_status_uninitialized_compiler_error;
+  }
+  auto* compiler = context->cross_compiler.get();
+
+  if (!binding_count) {
+    context->messages.append(
+        "Invoked needs_buffer_size_buffer without a valid binding_count "
```

Comment:
What is needs_buffer_size_buffer, should that be binding_info?

Reply:
Done